src/biokbase/narrative/jobs/job.py

```diff
@@ -341,16 +330,18 @@ def _update_state(self, state: dict) -> None:
         # Check if there would be no change in updating
         # i.e., if state <= self._acc_state
         if self._acc_state is not None:
-            if {**self._acc_state, **state} == self._acc_state:
+            if merge(self._acc_state, state) == self._acc_state:
                 return
 
         state = copy.deepcopy(state)
         if self._acc_state is None:
             self._acc_state = state
         else:
-            self._acc_state.update(state)
+            merge_inplace(self._acc_state, state)
+
+        self.last_updated = time.time_ns()
```

Comment:
can you alter this so that all jobs updated that are updated from a batch of ee2 data get the same timestamp?

Reply:
Curious ... why?
I was thinking about race conditions ...
I assuaged myself that all backend requests execute synchronously ...
The `ts`/`last_checked` and `last_updated` come from the backend so the STATUS requests are in effect atomic wrt to each other.
Am I missing something?

Reply:
I think if we just update `last_updated` when the jobs are updated from a batch of ee2 data though, there would be less chance of spam, because I'm actually hazy on the full behavior of the ee2 state and how it might spam `_acc_state` with updates

Reply:
I want to ensure that when the frontend receives a batch of job data, all will have the same timestamp. If the timestamp is applied individually within each job object using `time.time_ns()`, there's the possibility that timestamps will vary across a batch -- esp. if batches are large.